docs/shared/reference/project-configuration.md

```diff
@@ -158,6 +158,19 @@ Examples:
 
 Note the result value is hashed, so it is never displayed.
 
+_External Dependencies_
+
+This input is useful for ambiguous executors like `nx:run-commands`.
+
+Examples:
+
+- Specify that a target executor is not affected by any changes in installed packages
+- `{externalDependencies: []}`
+- Specify exact packages that affect a target executor
+- `{externalDependencies: ["lerna"]}`
+
+Note the result value is hashed, so it is never displayed.
```

Comment:
```suggestion
This input type should only be used with executors which run arbitrary commands/scripts without having a way of determining which, if any, external dependencies are associated with the target. A canonical example is the `nx:run-commands`. Because any command can be provided as an option, it is not possible to tell what the command does or what external dependencies it may use.

> Note: By default, Nx will hash _all_ external dependencies for targets that use `nx:run-commmands` and other unknown executors. This is the safest option because it ensures that any external dependencies are captured. This may lead to unexpected cache misses when updating dependencies. This option is a means to override this behavior and allow for more cache hits by providing exactly which dependencies are used by the target.

Examples:
- Specify that a target executor is not affected by any changes in installed packages
- `{externalDependencies: []}`
- Specify exact packages that affect a target executor
- `{externalDependencies: ["lerna"]}`

Note the result value is hashed, so it is never displayed.
```